boot/bootutil/src/boot_record.c

```diff
@@ -230,3 +225,105 @@ boot_save_boot_status(uint8_t sw_module,
     return 0;
 }
 #endif /* MCUBOOT_MEASURED_BOOT */
+
+#ifdef MCUBOOT_DATA_SHARING
+int boot_save_shared_data(const struct image_header *hdr, const struct flash_area *fap,
+                          const uint8_t slot, const int max_app_size)
+{
+    int rc;
+
+#if defined(MCUBOOT_SINGLE_APPLICATION_SLOT)
+    uint8_t mode = MCUBOOT_MODE_SINGLE_SLOT;
+#elif defined(MCUBOOT_SWAP_USING_SCRATCH)
+    uint8_t mode = MCUBOOT_MODE_SWAP_USING_SCRATCH;
+#elif defined(MCUBOOT_OVERWRITE_ONLY)
+    uint8_t mode = MCUBOOT_MODE_UPGRADE_ONLY;
+#elif defined(MCUBOOT_SWAP_USING_MOVE)
+    uint8_t mode = MCUBOOT_MODE_SWAP_USING_MOVE;
+#elif defined(MCUBOOT_DIRECT_XIP)
+    uint8_t mode = MCUBOOT_MODE_DIRECT_XIP;
+#elif defined(MCUBOOT_RAM_LOAD)
+    uint8_t mode = MCUBOOT_MODE_RAM_LOAD;
+#else
+#error "Unknown mcuboot operating mode"
+#endif
+
+#if defined(MCUBOOT_SIGN_RSA)
+    uint8_t signature_type = MCUBOOT_SIGNATURE_TYPE_RSA;
+#elif defined(MCUBOOT_SIGN_EC256)
+    uint8_t signature_type = MCUBOOT_SIGNATURE_TYPE_ECDSA_P256;
+#elif defined(MCUBOOT_SIGN_ED25519)
+    uint8_t signature_type = MCUBOOT_SIGNATURE_TYPE_ED25519;
+#else
+    uint8_t signature_type = MCUBOOT_SIGNATURE_TYPE_NONE;
+#endif
+
+#if defined(MCUBOOT_SERIAL)
+    uint8_t recovery = MCUBOOT_RECOVERY_MODE_SERIAL_RECOVERY;
+#elif defined(MCUBOOT_SERIAL_WAIT_FOR_DFU) || (defined(__ZEPHYR__) && defined(CONFIG_BOOT_USB_DFU_GPIO))
+    uint8_t recovery = MCUBOOT_RECOVERY_MODE_DFU;
```

Comment:
The use of `__zephyr__` in a bootutil file should not be allowed, a generic `CONFIG_...` define should be created to add support for this case.

Reply:
Honestly - even `CONFIG_...` shouldn't be used directly here. Rathe any Kconfig should be converted into Mcuboot-platform-agnostic configuration definition by `bootloader/mcuboot/boot/zephyr/include/mcuboot_config/mcuboot_config.h` and that one might be usede.

Reply:
Fixed